### PR TITLE
CLI Tool: Allow using `--file -` to read metadata from stdin

### LINKED ISSUE #1326: subxt codegen should support the input of metadata through stdin
That would allow to pipe it in the following manner:

`parachain-node export-metadata | subxt codegen --file - | rustfmt --edition=2021 --emit-stdout > metadata.rs`

### PR DESCRIPTION
**Pull request by tadeohepperle**
fixes #1326

I tested it manually by running `subxt metadata | subxt codegen --file -` (which is not super useful obviously) but shows that it works for the requested usecase of running `parachain-node export-metadata | subxt codegen --file -`.

### CODE REVIEW

File: cli/src/utils.rs [hunk 1]
```diff
@@ -33,6 +33,26 @@ pub struct FileOrUrl {
...
+#[derive(Debug, Clone)]
+pub enum PathOrStdIn {
+    Path(PathBuf),
+    StdIn,
+}
+
+impl FromStr for PathOrStdIn {
+    type Err = <PathBuf as FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```
Comment (niklasad1) [inline cli/src/utils.rs:47]: I wonder whether we should trim the leading and trailing whitespaces before checking that the `--file is `-`

Comment (niklasad1) [reply cli/src/utils.rs:47]: and a simple test for the `from_str` impl would be nice to have.

**Review by niklasad1 — APPROVED** [review]

File: cli/src/utils.rs (hunk 1 above)
Comment (tadeohepperle) [reply cli/src/utils.rs:47]: I think regarding trimming, `clap` already takes care of that before the args are given to us to parse, but I am not entirely sure.

Comment (niklasad1) [reply cli/src/utils.rs:69]: Tadeo added trimming and tests for it :)

**Review by jsdw — APPROVED** [review]
LGTM, nice one!

**Pull request merged**